feat(perl): Added perl version module (#1382)
starship/starship: docs/config/README.md

symbol = "🐪 "
```

## Perl

The `perl` module shows the currently installed version of Perl.
The module will be shown if any of the following conditions are met:

> - The current directory contains a `Makefile.PL` file

Reviewer: There is also `Build.PL` used by [`Module::Build`](https://metacpan.org/pod/Module::Build) and [`Module::Build::Tiny`](https://metacpan.org/pod/Module::Build::Tiny).

Author: added, doc updated

> - The current directory contains a `cpanfile` file

Reviewer: Well, if you support `cpanfile` then you should probably support  `cpanfile.snapshot` too. That's pretty much CPAN's version of a lockfile.

Author: added, doc updated, though i think there has to be a cpanfile if there is a cpanfile.snapshot, not sure, it wont hurt anyhow.

Reviewer: > added, doc updated, though i think there has to be a cpanfile if there is a cpanfile.snapshot, not sure, it wont hurt anyhow.

That would be pretty unusual, I agree... But as you said, won't hurt 🙃

- The current directory contains a `META.json` file
- The current directory contains a `META.yml` file
- The current directory contains a `.perl-version` file
- The current directory contains a `.pl` file

> - The current directory contains a `.pm` file

Reviewer: There are also documentation files aka `.pod` files.

Author: added .pod file

Author: added, doc updated

### Options

| Variable   | Default      | Description                                           |
| ---------- | ------------ | ----------------------------------------------------- |
| `symbol`   | `"🐪 "`      | The symbol used before displaying the version of PHP. |
| `style`    | `"bold 149"` | The style for the module.                             |
| `disabled` | `false`      | Disables the `perl` module.                           |

### Example

```toml
# ~/.config/starship.toml

[perl]

> symbol = "🐫 "

Reviewer: Nasty, I'm sure OCaml folks would loudly object to the abuse of their camel 😀

Author: this was a joke, as just the same thing is done a few lines up, i replace example by an oyster

Reviewer: My comment was also a joke 🤣

format = "via [🐪 $version]($style) "
```

## Perl

The `perl` module shows the currently installed version of Perl.
The module will be shown if any of the following conditions are met:

> - The current directory contains a `Makefile.PL` file
> - The current directory contains a `Build.PL` file
> - The current directory contains a `cpanfile` file
> - The current directory contains a `cpanfile.snapshot` file
> - The current directory contains a `META.json` file
> - The current directory contains a `META.yml` file
> - The current directory contains a `.perl-version` file
> - The current directory contains a `.pl` file
> - The current directory contains a `.pm` file

Reviewer: ```suggestion
- The current directory contains a `Makefile.PL` or `Build.PL` file
- The current directory contains a `cpanfile` or `cpanfile.snapshot` file
- The current directory contains a `META.json` or `META.yml` file
- The current directory contains a `.perl-version` file
- The current directory contains a file with `.pl`, `.pm` or `.pod` extension
```

Reviewer: It is mostly a cosmetic suggestion for sake of decreasing the number of lines.

- The current directory contains a `.pod` file

### Options

| Variable   | Default      | Description                                           |
| ---------- | ------------ | ----------------------------------------------------- |

> | `symbol`   | `"🐪 "`      | The symbol used before displaying the version of PHP. |

Reviewer: You have PHP leftover here.

Author: oups

Reviewer: Mah, it happens all the time 😃

| `style`    | `"bold 149"` | The style for the module.                             |
| `disabled` | `false`      | Disables the `perl` module.                           |

### Example

```toml
# ~/.config/starship.toml

[perl]
symbol = "🦪 "

> ```

Reviewer: ### Options

| Option     | Default                            | Description                                             |
| ---------- | ---------------------------------- | ------------------------------------------------------- |
| `format`   | `"via [$symbol$version]($style) "` | The format string for the module.                       |
| `symbol`   | `"🐪 "`                            | The symbol used before displaying the version of Perl. |
| `style`    | `"bold 149"`                    | The style for the module.                               |
| `disabled` | `false`                            | Disables the `perl` module.                            |

### Variables

| Variable | Example   | Description                          |
| -------- | --------- | ------------------------------------ |
| version  | `v5.26.1` | The version of `perl`               |
| symbol   |           | Mirrors the value of option `symbol` |
| style\*  |           | Mirrors the value of option `style`  |

\*: This variable can only be used as a part of a style string

### Example

```toml
# ~/.config/starship.toml

[perl]
format = "via [🦪 $version]($style) "
```

format = "via [🐪 $version]($style) "
```

## Perl

The `perl` module shows the currently installed version of Perl.
The module will be shown if any of the following conditions are met:

- The current directory contains a `Makefile.PL` or `Build.PL` file
- The current directory contains a `cpanfile` or `cpanfile.snapshot` file
- The current directory contains a `META.json` file or `META.yml` file
- The current directory contains a `.perl-version` file
- The current directory contains a `.pl`, `.pm` or `.pod`

### Options

| Variable   | Default                            | Description                                           |
| ---------- |----------------------------------- | ----------------------------------------------------- |

> | `format`   | `"via [$symbol$version]($style) "` | The symbol used before displaying the version of Perl |

Reviewer: I'm sorry 🙃 
```suggestion
| `format`   | `"via [$symbol$version]($style) "` | The format string for the module.                       |
```
